frontend/javascripts/oxalis/model/sagas/quick_select_ml_saga.ts

```diff
@@ -0,0 +1,279 @@
+  // Somewhere between the front-end, the back-end and the embedding
+  // server, there seems to be a different linearization of 2D the image
+  // data which is why the code here deals with the XZ plane as a special
+  // case.
+  const stride =
+    activeViewport === "PLANE_XZ"
+      ? [size[1], size[0], size[0] * size[1] * size[2]]
+      : [size[2], size[0], size[0] * size[1] * size[2]];
+
+  let mask = ndarray(maskData, size, stride);
+  mask = mask
+    // a.lo(x,y) => a[x:, y:]
+    .lo(topLeft[firstDim], topLeft[secondDim], 0)
+    // a.hi(x,y) => a[:x, :y]
+    .hi(userSizeInTargetMag[firstDim], userSizeInTargetMag[secondDim], 1);
+  return mask;
+}
+
+export function* prefetchEmbedding(action: MaybePrefetchEmbeddingAction) {
+  const preparation = yield* call(prepareQuickSelect, action, true);
+  if (preparation == null) {
+    return;
+  }
+  const { labeledResolution, activeViewport, colorLayer } = preparation;
+  const { startPosition } = action;
+  const PREFETCH_WINDOW_SIZE = [100, 100, 0] as Vector3;
+  const endPosition = V3.add(
+    startPosition,
+    Dimensions.transDim(PREFETCH_WINDOW_SIZE, activeViewport),
+  );
+
+  const userBoxMag1 = new BoundingBox({
+    min: V3.floor(V3.min(startPosition, endPosition)),
+    max: V3.floor(
+      V3.add(V3.max(startPosition, endPosition), Dimensions.transDim([0, 0, 1], activeViewport)),
+    ),
+  });
+
+  console.time("prefetch session and embedding");
+  const dataset = yield* select((state: OxalisState) => state.dataset);
+  // Won't block, because the return value is not a promise (but contains
+  // a promise instead)
+  const { embeddingPromise } = yield* call(
+    getEmbedding,
+    dataset,
+    colorLayer.name,
+    userBoxMag1,
+    labeledResolution,
+    activeViewport,
+  );
+  // Also prefetch session (will block). After the first time, it's basically
+  // a noop.
+  yield* call(getSession);
+
+  try {
+    // Await the promise here so that the saga finishes once the embedding was loaded
+    // (this simplifies debugging and time measurement).
+    yield* call(() => embeddingPromise);
+  } catch (exception) {
+    console.error(exception);
+    // Don't notify user because we are only prefetching.
+  }
+  console.timeEnd("prefetch session and embedding");
+}
+
+export default function* performQuickSelect(action: ComputeQuickSelectForRectAction): Saga<void> {
+  const preparation = yield* call(prepareQuickSelect, action, true);
+  if (preparation == null) {
+    return;
+  }
+  const {
+    labeledZoomStep,
+    labeledResolution,
+    firstDim,
+    secondDim,
+    thirdDim,
+    activeViewport,
+    volumeTracing,
+    colorLayer,
+  } = preparation;
+  const { startPosition, endPosition, quickSelectGeometry } = action;
+
+  const userBoxMag1 = new BoundingBox({
+    min: V3.floor(V3.min(startPosition, endPosition)),
+    max: V3.floor(
+      V3.add(V3.max(startPosition, endPosition), Dimensions.transDim([0, 0, 1], activeViewport)),
```

Comment:
Here, I understand where the V3.min and V3.max is coming from, because the user could've created the bounding box from right to left/bottom to top.